starfyre/parser.py

```diff
@@ -166,11 +170,25 @@ def handle_endtag(self, tag):
         if endtag_node.tag != "style" and endtag_node.tag != "script":
             if len(self.stack) > 0:
                 parent_node = self.stack[-1]      #this is last item/"top element" of stack
-                if parent_node.original_name == endtag_node.original_name:
-                    parent_node.children.extend(self.current_children)
-                    self.current_children = []
-                else:
-                    self.current_children.append(endtag_node)
+                if endtag_node.original_name != "": #this node is for "custom" component, so we should check for <slot> tags 
```

Comment:
`original_name` should never be empty. If we have a `div` element then the original name should be `div`. It is empty right now as it is a bug and should be addressed